### PR TITLE
DFC 1.6 prototype

### LINKED ISSUE #9170: OFN DFC Products
## Description

I want to be able use the DFC standard to interface with OFN.

The OFN v0 Products endpoint gives enough functionality for a simple interface with the DFC. By linking OFN v0 product endpoint to the DFC we can build integrations from the DFC standard and then change OFN under the hood without impacting the integrations. As the v0 products endpoint has everything that is needed, this will mean that we can deliver on DFC requirements and AU API requirements and build integrations that will be supported while we make major changes to the OFN product model.

Another advantage is that we'll minimise some of the risk of using the OFN v0 endpoint. For example some fields if updated via the endpoint may break existing products -> updating unit_type of a variant can break other products. Due to the split between product and variant models in OFN updating fields on the product can have adverse effects. This caused a lot of headache when writing product import, so any update functionality via the API will run the same risks. We'll be looking at this in #9069.

This is a proposal, and currently a placeholder. If we have @mkllnk's buy in to the approach we'll set up a kick off with Garethe and Maikel and others to get this work started.

The exact data fields and API functions we'll be serving.... coming soon!
Basic CRUD but with a more limited set of fields. 

TODO - Understand how we are going to map OFN producer price to the DFC model.

## Acceptance Criteria & Tests

1. A DFC Products compliant endpoint that serves OFN data.
2. The DFC connector should be independent of OFN code, and should interface with the OFN v0 API

### PR DESCRIPTION
**Pull request by mkllnk**
### ~~WIP~~

* ~~Keycloak currently doesn't verify the email address and we don't check that either. So currently you can hijack someone's OFN catalog by signing up at Keycloak with their email address unless it's registered already. We need to check against "uid" instead of "email".~~

#### What? Why?

- Closes #9170  

This PR also includes #9889 and supersedes it, I guess. I would have split this in parts but I don't have time to wait for several PRs to go through the pipe. So here is the full demo.

#### What should we test?

- Have an account on https://staging.coopcircuits.fr/ with products (as producer).
- Visit http://proto.datafoodconsortium.org/
- Login with your OFN email address you also have on the French staging server.
- Import product data.
- Change data and upload back to OFN.
- Check in your OFN account that the product name updated.

#### Release notes

Changelog Category: User facing changes | Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by mkllnk** [comment]
@dacook I think that the only fix needed to solve the authentication issue is this code change plus adjusting specs:
```diff
diff --git a/engines/dfc_provider/app/services/dfc_provider/authorization_control.rb b/engines/dfc_provider/app/services/dfc_provider/authorization_control.rb
index a7da60984..18ba0c410 100644
--- a/engines/dfc_provider/app/services/dfc_provider/authorization_control.rb
+++ b/engines/dfc_provider/app/services/dfc_provider/authorization_control.rb
@@ -48,7 +48,9 @@ module DfcProvider
     end
 
     def find_ofn_user(payload)
-      Spree::User.find_by(email: payload["email"])
+      return if payload["email"].blank?
+
+      Spree::User.find_by(uid: payload["email"])
     end
   end
 end
```

**Comment by dacook** [comment]
I've rebased, and added Maikel's suggested code, with some minor test updates to cater for it (and accidentally pushed it already). 
But I can't say understand it well enough to be confident to submit the PR myself yet. Based on my limited understanding so far, I think we should also add a test for an invalid or nil email/UID.

**Review by jibees — APPROVED** [review]
I've reviewed your commits @dacook and it looks good to me! 👌

**Comment by sigmundpetersen** [comment]
I think we can move this on with the 1 review.
What do you say @RachL ?

**Comment by RachL** [comment]
@sigmundpetersen yes but I think @dacook wanted to do a last round before? as per this comment: 

> I'd like to go through this one more time for a code review before approving it myself though, as I'm not sure if I understood it all the first time around.

**Comment by RachL** [comment]
testing is blocked for now: DFC prototype is experiencing issues.
